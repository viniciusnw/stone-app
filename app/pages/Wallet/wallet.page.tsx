import React from "react";
import { Text, View } from "react-native";
import {
  CreditCard,
  Profile as ProfileComp,
  Wallet as WalletComp,
} from "@Components";
import { useNavigation } from "@react-navigation/native";

export type CardProps = {
  name: string;
  number: string;
  brand: "visa" | "mastercard" | "santander";
};

export default function Wallet(props: any) {
  const navigation = useNavigation();

  const user = {
    src: "https://i.pinimg.com/280x280_RS/39/fd/09/39fd09a3f0962e5b7b4a3ccbf5a61057.jpg",
    name: "Bruce Wayne",
    rule: "Wayne Interprise",
  };

  const card: CardProps = {
    name: "Bruce Wayne",
    number: "**** **** **** 1234",
    brand: "visa",
  };

  const card1: CardProps = {
    name: "Dick Grayson",
    number: "**** **** **** 1234",
    brand: "mastercard",
  };

  React.useEffect(
    () =>
      props.setState({
        bottomBar: {
          menu: [
            {
              icon: "wallet",
              label: "Carteira",
              onPress: () => true,
              primary: false,
            },
            {
              icon: "home",
              label: "Home",
              onPress: () => props.goTo("Home"),
              primary: true,
            },
            {
              icon: "logout",
              label: "Logout",
              onPress: () => true,
              primary: false,
            },
          ],
        },
        topBar: {
          title: "Wallet",
          back: navigation.goBack,
        },
      }),
    []
  );

  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <WalletComp value="R$ 10.000,00" qty={200} />

      <View style={{ padding: 10 }}>
        <Text style={{ marginBottom: 15 }}>Meus cartoes:</Text>
        <CreditCard {...card} />
        <CreditCard {...card1} />
      </View>
    </View>
  );
}
