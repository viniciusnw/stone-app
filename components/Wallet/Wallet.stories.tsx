import React from "react";
import { Wallet, StorybookWrapper } from "@Components";


const sample = {
  value: "R$1350,50",
  qty: 8,
};

export const Default = {};

export default {
  title: "Wallet",
  component: (props: any) => (
    <StorybookWrapper>
      <Wallet {...sample} />
    </StorybookWrapper>
  ),
};
