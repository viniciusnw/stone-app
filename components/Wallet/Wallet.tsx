import React from 'react';
import styled from 'styled-components/native';
import { View, Text } from 'react-native';
import Icon from '@Components/Icon';

const StyledContainer = styled(View)`
  width: 100%;
  align-items: center;
  padding: 60px ${({ theme }) => theme.spacing.regular};
`;

const StyledValue = styled(Text)`
  font-size: 32px;
  font-weight: 700;
  color: ${({ theme }) => theme.color.black};
`;

const StyledQty = styled(Text)`
  margin-top: 5px;
  font-weight: 700;
  color: ${({ theme }) => theme.color.gray};
`;

const StyledIcon = styled(View)`
  width: 52px;
  height: 52px;
  align-items: center;
  justify-content: center;
  border: 1px solid ${({ theme }) => theme.color.border};
  border-radius: 100px;
  margin-bottom: ${({ theme }) => theme.spacing.regular};
`;

const Wallet = ({ value, qty, ...props }: any) => {
  return (
    <StyledContainer {...props}>
      <StyledIcon>
        <Icon icon="wallet" size={26} />
      </StyledIcon>
      <StyledValue>{value}</StyledValue>
      <StyledQty>{`${qty} pedido${qty > 1 ? 's' : ''}`}</StyledQty>
    </StyledContainer>
  );
};

export default Wallet;
