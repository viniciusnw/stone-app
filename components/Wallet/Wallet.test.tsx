import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Wallet from "./Wallet";

jest.useFakeTimers();

const sample = {
  value: "R$1350,50",
  qty: 8,
};

test("Wallet", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Wallet {...sample} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
