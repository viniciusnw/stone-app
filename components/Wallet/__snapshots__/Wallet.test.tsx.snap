// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "paddingBottom": 60,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 60,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomLeftRadius": 100,
        "borderBottomRightRadius": 100,
        "borderColor": "#ddd",
        "borderStyle": "solid",
        "borderTopLeftRadius": 100,
        "borderTopRightRadius": 100,
        "borderWidth": 1,
        "height": 52,
        "justifyContent": "center",
        "marginBottom": 15,
        "width": 52,
      }
    }
  >
    <Text />
  </View>
  <Text
    style={
      {
        "color": "#111111",
        "fontSize": 32,
        "fontWeight": "700",
      }
    }
  >
    R$1350,50
  </Text>
  <Text
    style={
      {
        "color": "#666666",
        "fontWeight": "700",
        "marginTop": 5,
      }
    }
  >
    8 pedidos
  </Text>
</View>
`;
